### PR TITLE
Manually update grakn to next release 0.17.1

### PR DESCRIPTION
**Pull request by fppt**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Manually update Grakn to next release 0.17.0. A manual update is required because a startup script change led to the automatic update: https://github.com/Homebrew/homebrew-core/pull/20198 failing

### CODE REVIEW

**Comment by ilovezfs** [comment]
@fppt what you have is the same as what I was testing locally that was not working. Did you actually try running `grakn server start`?

**Comment by MikeMcQuaid** [comment]
@fppt What is a sensible default string to set `CASSANDRA_HOME` to? The formula as-is seems to require users to set it but provides no guidance as to what it should be and why. Thanks!

**Comment by fppt** [comment]
Hi all apologies for the confusion. The brew formulae of Grakn was originally created by someone else so in my case I kind of had to muck around with it till it worked. I think it's in working order now.

**Comment by smartinsightsfromdata** [comment]
@ilovezfs @MikeMcQuaid  Good to go?
Thanks

**Comment by ilovezfs** [comment]
> Good to go?

Nope:

```
iMac-TMP:~ joe$ grakn server status
Grakn: Database For AI

Storage: NOT RUNNING
Queue: NOT RUNNING
Grakn: NOT RUNNING
iMac-TMP:~ joe$ grakn server start
Grakn: Database For AI

Starting Storage...FAILED!
Unable to start Storage
Starting Queue.../usr/local/Cellar/grakn/0.17.0/libexec/services/cassandra/grakn: line 179: /usr/local/Cellar/grakn/0.17.0/libexec/services/cassandra/services/redis/redis-server-osx: No such file or directory
FAILED!
Unable to start Queue
Starting Grakn...............
```
and  Mike's question about `CASSANDRA_HOME` hasn't been answered.

**Comment by fppt** [comment]
Yet another try. 

@ilovezfs @MikeMcQuaid I believe `CASSANDRA_HOME` now `SERVICE_HOME` is just used to set the paths needed by the startup script. 

The default is determined by where the package is installed which is why it gets piped in during install. I am open to improvements on this formulae if you guys have any suggestions. 

As I said it was created by someone else and I am still trying to maintain it and update it.

**Comment by ilovezfs** [comment]
What should be the default value of SERVICE_HOME?

**Comment by fppt** [comment]
From the root of the package `services/`

**Comment by ilovezfs** [comment]
Currently, this is what's getting installed:
```
iMac-TMP:homebrew-core joe$ cat `which grakn`
#!/bin/bash
SERVICE_HOME="" exec "/usr/local/Cellar/grakn/0.17.0/libexec/grakn" "$@"
iMac-TMP:homebrew-core joe$
```
I don't think we want to be forcibly unsetting the variable in a wrapper script, so maybe use `bin.install_symlink` instead.

Also it's still not starting.

**Comment by ilovezfs** [comment]
```
iMac-TMP:libexec joe$ cat logs/grakn.log 
2017-11-06 04:40:14,059 [main] INFO  ai.grakn.engine.GraknEngineConfig - Project directory in use: /usr/local/Cellar/grakn/0.17.0/libexec/services/
2017-11-06 04:40:14,062 [main] INFO  ai.grakn.engine.GraknEngineConfig - Configuration file in use: /usr/local/Cellar/grakn/0.17.0/libexec/conf/grakn.properties
2017-11-06 04:40:14,062 [main] INFO  ai.grakn.engine.GraknEngineConfig - Number of threads set to: 4
2017-11-06 04:40:14,196 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:14,241 [main] INFO  a.g.e.t.m.r.RedisTaskManager - Running queue consumer with 32 execution threads
2017-11-06 04:40:14,372 [main] INFO  ai.grakn.engine.GraknEngineServer - Starting task manager ai.grakn.engine.tasks.manager.redisqueue.RedisTaskManager
2017-11-06 04:40:14,379 [main] INFO  ai.grakn.engine.GraknEngineServer - 
==================================================
2017-11-06 04:40:14,379 [main] INFO  ai.grakn.engine.GraknEngineServer - 
     ___  ___  ___  _  __ _  _     ___  ___     
    / __|| _ \/   \| |/ /| \| |   /   \|_ _|    
   | (_ ||   /| - || ' < | .` | _ | - | | |     
    \___||_|_\|_|_||_|\_\|_|\_|(_)|_|_||___|   

 Web Dashboard available at [http://0.0.0.0:4567]
2017-11-06 04:40:14,379 [main] INFO  ai.grakn.engine.GraknEngineServer - 
==================================================
2017-11-06 04:40:14,381 [main] INFO  ai.grakn.engine.GraknEngineServer - iMac-TMP.local-3ea400e7-4f38-4edd-aab5-aa3f6fdd4dc2 is checking the system schema
2017-11-06 04:40:16,844 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:19,453 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:22,097 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:24,749 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:27,364 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:29,995 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:32,988 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:34,604 [main] ERROR ai.grakn.engine.SystemKeyspace - Error while loading system schema in 20.22 s. The error was: Could not instantiate implementation: org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager
java.lang.IllegalArgumentException: Could not instantiate implementation: org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager
	at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:69)
	at org.janusgraph.diskstorage.Backend.getImplementationClass(Backend.java:477)
	at org.janusgraph.diskstorage.Backend.getStorageManager(Backend.java:409)
	at org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.<init>(GraphDatabaseConfiguration.java:1353)
	at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:107)
	at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:97)
	at org.janusgraph.core.JanusGraphFactory$Builder.open(JanusGraphFactory.java:152)
	at ai.grakn.factory.TxFactoryJanus.configureGraph(TxFactoryJanus.java:173)
	at ai.grakn.factory.TxFactoryJanus.newJanusGraph(TxFactoryJanus.java:127)
	at ai.grakn.factory.TxFactoryJanus.buildTinkerPopGraph(TxFactoryJanus.java:123)
	at ai.grakn.factory.TxFactoryJanus.buildTinkerPopGraph(TxFactoryJanus.java:72)
	at ai.grakn.factory.TxFactoryAbstract.getTinkerPopGraph(TxFactoryAbstract.java:121)
	at ai.grakn.factory.TxFactoryAbstract.getTinkerPopGraph(TxFactoryAbstract.java:115)
	at ai.grakn.factory.TxFactoryAbstract.getGraph(TxFactoryAbstract.java:96)
	at ai.grakn.factory.TxFactoryAbstract.open(TxFactoryAbstract.java:83)
	at ai.grakn.engine.factory.EngineGraknTxFactory.tx(EngineGraknTxFactory.java:78)
	at ai.grakn.engine.SystemKeyspace.loadSystemSchema(SystemKeyspace.java:164)
	at ai.grakn.engine.GraknEngineServer.loadAndUnlock(GraknEngineServer.java:205)
	at ai.grakn.engine.GraknEngineServer.lockAndInitializeSystemSchema(GraknEngineServer.java:193)
	at ai.grakn.engine.GraknEngineServer.start(GraknEngineServer.java:163)
	at ai.grakn.engine.GraknEngineServer.main(GraknEngineServer.java:144)
Caused by: java.lang.reflect.InvocationTargetException: null
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488)
	at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:58)
	... 20 common frames omitted
Caused by: org.janusgraph.diskstorage.TemporaryBackendException: Temporary failure in storage backend
	at org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager.ensureKeyspaceExists(AstyanaxStoreManager.java:601)
	at org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager.<init>(AstyanaxStoreManager.java:313)
	... 25 common frames omitted
Caused by: com.netflix.astyanax.connectionpool.exceptions.PoolTimeoutException: PoolTimeoutException: [host=127.0.0.1(127.0.0.1):9160, latency=10002(10002), attempts=1]Timed out waiting for connection
	at com.netflix.astyanax.connectionpool.impl.SimpleHostConnectionPool.waitForConnection(SimpleHostConnectionPool.java:231)
	at com.netflix.astyanax.connectionpool.impl.SimpleHostConnectionPool.borrowConnection(SimpleHostConnectionPool.java:198)
	at com.netflix.astyanax.connectionpool.impl.RoundRobinExecuteWithFailover.borrowConnection(RoundRobinExecuteWithFailover.java:84)
	at com.netflix.astyanax.connectionpool.impl.AbstractExecuteWithFailoverImpl.tryOperation(AbstractExecuteWithFailoverImpl.java:117)
	at com.netflix.astyanax.connectionpool.impl.AbstractHostPartitionConnectionPool.executeWithFailover(AbstractHostPartitionConnectionPool.java:352)
	at com.netflix.astyanax.thrift.ThriftClusterImpl.executeSchemaChangeOperation(ThriftClusterImpl.java:146)
	at com.netflix.astyanax.thrift.ThriftClusterImpl.internalCreateKeyspace(ThriftClusterImpl.java:321)
	at com.netflix.astyanax.thrift.ThriftClusterImpl.addKeyspace(ThriftClusterImpl.java:294)
	at org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager.ensureKeyspaceExists(AstyanaxStoreManager.java:596)
	... 26 common frames omitted
2017-11-06 04:40:34,607 [main] ERROR ai.grakn.engine.GraknEngineServer - An exception has occurred
java.lang.IllegalArgumentException: Could not instantiate implementation: org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager
	at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:69)
	at org.janusgraph.diskstorage.Backend.getImplementationClass(Backend.java:477)
	at org.janusgraph.diskstorage.Backend.getStorageManager(Backend.java:409)
	at org.janusgraph.graphdb.configuration.GraphDatabaseConfiguration.<init>(GraphDatabaseConfiguration.java:1353)
	at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:107)
	at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:97)
	at org.janusgraph.core.JanusGraphFactory$Builder.open(JanusGraphFactory.java:152)
	at ai.grakn.factory.TxFactoryJanus.configureGraph(TxFactoryJanus.java:173)
	at ai.grakn.factory.TxFactoryJanus.newJanusGraph(TxFactoryJanus.java:127)
	at ai.grakn.factory.TxFactoryJanus.buildTinkerPopGraph(TxFactoryJanus.java:123)
	at ai.grakn.factory.TxFactoryJanus.buildTinkerPopGraph(TxFactoryJanus.java:72)
	at ai.grakn.factory.TxFactoryAbstract.getTinkerPopGraph(TxFactoryAbstract.java:121)
	at ai.grakn.factory.TxFactoryAbstract.getTinkerPopGraph(TxFactoryAbstract.java:115)
	at ai.grakn.factory.TxFactoryAbstract.getGraph(TxFactoryAbstract.java:96)
	at ai.grakn.factory.TxFactoryAbstract.open(TxFactoryAbstract.java:83)
	at ai.grakn.engine.factory.EngineGraknTxFactory.tx(EngineGraknTxFactory.java:78)
	at ai.grakn.engine.SystemKeyspace.loadSystemSchema(SystemKeyspace.java:164)
	at ai.grakn.engine.GraknEngineServer.loadAndUnlock(GraknEngineServer.java:205)
	at ai.grakn.engine.GraknEngineServer.lockAndInitializeSystemSchema(GraknEngineServer.java:193)
	at ai.grakn.engine.GraknEngineServer.start(GraknEngineServer.java:163)
	at ai.grakn.engine.GraknEngineServer.main(GraknEngineServer.java:144)
Caused by: java.lang.reflect.InvocationTargetException: null
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:488)
	at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:58)
	... 20 common frames omitted
Caused by: org.janusgraph.diskstorage.TemporaryBackendException: Temporary failure in storage backend
	at org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager.ensureKeyspaceExists(AstyanaxStoreManager.java:601)
	at org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager.<init>(AstyanaxStoreManager.java:313)
	... 25 common frames omitted
Caused by: com.netflix.astyanax.connectionpool.exceptions.PoolTimeoutException: PoolTimeoutException: [host=127.0.0.1(127.0.0.1):9160, latency=10002(10002), attempts=1]Timed out waiting for connection
	at com.netflix.astyanax.connectionpool.impl.SimpleHostConnectionPool.waitForConnection(SimpleHostConnectionPool.java:231)
	at com.netflix.astyanax.connectionpool.impl.SimpleHostConnectionPool.borrowConnection(SimpleHostConnectionPool.java:198)
	at com.netflix.astyanax.connectionpool.impl.RoundRobinExecuteWithFailover.borrowConnection(RoundRobinExecuteWithFailover.java:84)
	at com.netflix.astyanax.connectionpool.impl.AbstractExecuteWithFailoverImpl.tryOperation(AbstractExecuteWithFailoverImpl.java:117)
	at com.netflix.astyanax.connectionpool.impl.AbstractHostPartitionConnectionPool.executeWithFailover(AbstractHostPartitionConnectionPool.java:352)
	at com.netflix.astyanax.thrift.ThriftClusterImpl.executeSchemaChangeOperation(ThriftClusterImpl.java:146)
	at com.netflix.astyanax.thrift.ThriftClusterImpl.internalCreateKeyspace(ThriftClusterImpl.java:321)
	at com.netflix.astyanax.thrift.ThriftClusterImpl.addKeyspace(ThriftClusterImpl.java:294)
	at org.janusgraph.diskstorage.cassandra.astyanax.AstyanaxStoreManager.ensureKeyspaceExists(AstyanaxStoreManager.java:596)
	... 26 common frames omitted
2017-11-06 04:40:35,665 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:38,298 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:40,936 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:43,569 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:46,186 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:48,799 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:51,450 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:54,082 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:56,731 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:40:59,377 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:01,994 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:04,640 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:07,264 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:09,874 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:12,487 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:15,136 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:17,747 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:20,380 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:23,017 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:25,654 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:28,300 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:30,939 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:33,565 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:36,182 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:38,800 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:41,439 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:44,049 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:46,708 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:49,338 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:51,979 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:54,614 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:57,226 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:41:59,872 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:42:02,486 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:42:05,116 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:42:07,762 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:42:10,401 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:42:13,042 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
2017-11-06 04:45:32,189 [main] INFO  ai.grakn.client.Client - Server 0.0.0.0:4567 is not running
iMac-TMP:libexec joe$ 
```

**Comment by fppt** [comment]
There is a know bug that if you start it from a directory with lots of files it may sometimes stall. I think you are experiencing that bug.

> I don't think we want to be forcibly unsetting the variable in a wrapper script, so maybe use `bin.install_symlink` instead.

Ya that is a much more elegant solution but our current startup script has an error in it preventing that. We will try to release a hot fix to repair that at some point.

**Comment by ilovezfs** [comment]
@fppt feel free to change the test to actually start the server if you think this is specific to my system.

**Comment by idealley** [comment]
Any news on this?

**Comment by fppt** [comment]
We are fixing the problem upstream which is preventing a startup on some systems. Once that problem has been fixed we will release a hot fix and then I will try to get this PR updated and working.

File: Formula/grakn.rb [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
   bottle :unneeded
 
   depends_on :java => "1.8+"
 
   def install
     libexec.install Dir["*"]
-    bin.install Dir["#{libexec}/bin/*"]
-    bin.env_script_all_files(libexec/"bin", :CASSANDRA_HOME => ENV["CASSANDRA_HOME"])
+    bin.install Dir["#{libexec}/grakn", "#{libexec}/graql"]
+    bin.env_script_all_files(libexec/"", :SERVICE_HOME => ENV["SERVICE_HOME"])
```
Comment (ilovezfs) [inline Formula/grakn.rb]: `SERVICE_HOME` needs to have a default value set in the formula or we should just be using
```
libexec.install Dir["*"]
bin.write_exec_script libexec/"grakn", libexec/"graql"
```

**Comment by ilovezfs** [comment]
The problem here is that it's not compatible with Java 9. The "better" test succeeds locally for me when I use Java 8. Have you tested with Java 9 yet locally?

**Comment by fppt** [comment]
I was wondering why the better test was failing. Thank you for the clarification @ilovezfs. No we have not tested with java 9. I don't think at this stage we can support java 9.

**Comment by ilovezfs** [comment]
OK, then we should force it to use Java 8.

```
    bin.install libexec/"bin/grakn", libexec/"bin/graql"
    bin.env_script_all_files(libexec/"bin", Language::Java.java_home_env("1.8"))
```

And make it `depends_on :java => "1.8"` instead of `depends_on :java => "1.8+"`

Then it should pass with the start/stop test.

**Comment by fppt** [comment]
When I add `bin.env_script_all_files(libexec, Language::Java.java_home_env("1.8"))` I am getting a permission issue locally:

```
Filipes-MacBook-Pro:~ filipe$ grakn server start
/usr/local/bin/grakn: line 2: /usr/local/Cellar/grakn/0.17.1/libexec/grakn: Permission denied
/usr/local/bin/grakn: line 2: exec: /usr/local/Cellar/grakn/0.17.1/libexec/grakn: cannot execute: Undefined error: 0
```

**Comment by ilovezfs** [comment]
```
diff --git a/Formula/grakn.rb b/Formula/grakn.rb
index 8199b60..0fdbaad 100644
--- a/Formula/grakn.rb
+++ b/Formula/grakn.rb
@@ -6,12 +6,12 @@ class Grakn < Formula
 
   bottle :unneeded
 
-  depends_on :java => "1.8+"
+  depends_on :java => "1.8"
 
   def install
     libexec.install Dir["*"]
-    bin.install Dir["#{libexec}/grakn", "#{libexec}/graql"]
-    bin.env_script_all_files(libexec/"", :SERVICE_HOME => ENV["SERVICE_HOME"])
+    bin.install libexec/"grakn", libexec/"graql"
+    bin.env_script_all_files(libexec, Language::Java.java_home_env("1.8"))
   end
 
   test do
```

**Comment by ilovezfs** [comment]
Looks like the test is trying to violate the sandbox by writing to `/usr/local/Cellar/grakn/0.17.1/libexec`. Can we have it storing the logs and data somewhere outside the Cellar? Usually `/usr/local/var/log` for logs, `/usr/local/var/lib/grakn` for data, and `/usr/local/var/run` for temporary state? Also, is there a way to have it temporarily use `testpath` for that stuff?

**Comment by ilovezfs** [comment]
```
==> Sandbox log
Nov 13 05:04:50 sandboxd[208]: java(71362) deny file-write-data /usr/local/Cellar/grakn/0.17.1/libexec/logs/cassandra.log
Nov 13 05:05:52 sandboxd[208]: redis-server-osx(72573) deny file-write-data /usr/local/Cellar/grakn/0.17.1/libexec/db/redis/appendonly.aof
Nov 13 05:06:05 sandboxd[208]: java(72639) deny file-write-data /usr/local/Cellar/grakn/0.17.1/libexec/logs/grakn.log
Nov 13 05:06:05 sandboxd[208]: java(72639) deny file-write-data /usr/local/Cellar/grakn/0.17.1/libexec/logs/grakn-postprocessing.log
```

**Comment by fppt** [comment]
> Can we have it storing the logs and data somewhere outside the Cellar? 

That will require us to make some more changes upstream. 

> Also, is there a way to have it temporarily use `testpath` for that stuff?

I am not sure maybe once the other changes are made we will be able to do that.

**Comment by ilovezfs** [comment]
OK, let's go ahead and use the simple test for now even though it didn't catch the Java 9 incompatibility. :see_no_evil:

**Comment by fppt** [comment]
Great! I have added the necessary path changes to our roadmap. So hopefully in the next release I can use the more thorough tests which won't leak into the Cellar.

**Comment by ilovezfs** [comment]
>Great! I have added the necessary path changes to our roadmap. So hopefully in the next release I can use the more thorough tests which won't leak into the Cellar.

Right, also you want the data to survive a `brew upgrade` … it's not just about the test :)

**Comment by fppt** [comment]
We don't actually support backwards compatibility at this stage but soon we making our first production ready release and indeed at that stage preserving the data will be very important.

I don't think many people would appreciate a DB which wipes your data with every release :)

**Comment by fppt** [comment]
With regards to this PR is it "good enough" to go in now? and then we can improve the formula further on the next release?

**Comment by ilovezfs** [comment]
@fppt thanks for your first contribution to Homebrew! We really appreciate upstream collaboration :)

**Comment by fppt** [comment]
Thanks @ilovezfs for all the assistance.

**Comment by ilovezfs** [comment]
@fppt you're welcome!

**Pull request merged**